panda/src/glstuff/glGraphicsStateGuardian_src.I

```diff
@@ -131,6 +131,39 @@ get_gl_version_minor() const {
 }
 
 /**
+ * Returns whether a core profile or a compatibility mode is considered.
+ */
+INLINE bool CLP(GraphicsStateGuardian)::
+has_core_profile() const {
+  return _core_profile;
+}
+
+/**
+ * Returns whether the fixed function pipeline is supported.
+ */
+INLINE bool CLP(GraphicsStateGuardian)::
+has_fixed_function_pipeline() const {
+#ifndef SUPPORT_FIXED_FUNCTION
+  return false;
+#else
+  // In OpenGL 3.1 the fixed function pipeline was declared deprecated, and
+  // theoretically all the pipeline should gone. However, some implementations
+  // kept them, "out of the law". After that, In OpenGL >= 3.2 the concept of
+  // core profile and compatibility mode where introduced. Therefore, since it
+  // is not possible to check whether those functions are implemented in GL 3.1
+  // or not, we should not try to use them.
+  if ((_gl_version_major == 3) && (_gl_version_minor == 1)) {
```

Comment:
That's a problem of OpenGL specification itself... For some reason they let the vendors to decide whether they are implementing fixed pipeline, but without providing a way to know it prior OpenGL >= 4... A bummer...
Theoretically, from the point of view of an engine built "from scratch", the safest way is just considering it is not available. However, as you are pointing out, this is not a new engine.

Along this line, I should say that I have not saw a single 3.1 platform without fixed pipeline... But I'm not an expert